From 1f178c14f8c33dddc71cf0c21c0d4c9ab8023554 Mon Sep 17 00:00:00 2001
From: Pavan S <psreenivas@habana.ai>
Date: Thu, 16 Mar 2023 13:01:41 +0200
Subject: [PATCH] [SW-136144] gaudi3: include specs_h9 in gaudi3 makefile

The SPECS_H9_DIR was added in common makefile which
is not correct. This patch moves it to gaudi3 makefile
and includes the gaudi3 makefile for compilation.

Signed-off-by: Pavan S <psreenivas@habana.ai>
Change-Id: Iaced3dbc4a909065232aa3d11c740a8d09484b91
---

diff --git a/Makefile b/Makefile
index 09148a1..ce0ac9d 100644
--- a/Makefile
+++ b/Makefile
@@ -18,9 +18,7 @@
 CFLAGS += -I$(SPECS_DIR)/$(BASE_DEV_NAME)/asic_reg
 CFLAGS += -I$(SPECS_EMBEDDED_DIR)
 CFLAGS += -I$(SPECS_EMBEDDED_DIR)/$(SPECS_EMB_DEV)
-CFLAGS += -I$(SPECS_H9_DIR)
 CFLAGS += -I ${EMBEDDED_ROOT}/arcboot/arch
-CFLAGS += -include $(SPECS_EMBEDDED_DIR)/$(SPECS_EMB_DEV)/$(SPECS_EMB_DEV)_reg_map.h
 CFLAGS += -include app/preboot/preboot_app_defs.h
 CFLAGS += -Wall -Werror
 CFLAGS += -g -c -fno-builtin
@@ -57,6 +55,7 @@
 PPBOOT_SRCS_C += $(shell ls app/ppboot/*.c)
 
 include $(HL_DEV_ARCH)/Makefile
+CFLAGS += -include $(SPECS_EMBEDDED_DIR)/$(SPECS_EMB_DEV)/$(SPECS_EMB_DEV)_reg_map.h
 
 PREBOOT_SRCS_C += $(shell ls $(HL_DEV_ARCH)/common.c)
 PREBOOT_SRCS_C += $(shell ls $(HL_DEV_ARCH)/$(HL_TARGET).c)
diff --git a/app/ext_rom/Makefile b/app/ext_rom/Makefile
index 40fb34c..543a6b8 100644
--- a/app/ext_rom/Makefile
+++ b/app/ext_rom/Makefile
@@ -7,10 +7,11 @@
 HL_DEV_ARCH_COMMON := $(PREBOOT_ROOT_DIR)/arch/$(BASE_DEV_NAME)/common
 SPECS_DEV := $(shell echo $(HL_DEV) | cut -d'_' -f1)
 
+include $(HL_DEV_ARCH_COMMON)/Makefile
+
 CFLAGS += -I$(SPECS_DIR)
 CFLAGS += -I$(SPECS_EMBEDDED_DIR)
 CFLAGS += -I$(SPECS_DIR)/$(BASE_DEV_NAME)/asic_reg
-CFLAGS += -I$(SPECS_H9_DIR)
 CFLAGS += -Wall -Werror
 CFLAGS += -g -c
 CFLAGS += -I $(PREBOOT_ROOT_DIR)/include/ -I $(PREBOOT_ROOT_DIR)/ -I $(PREBOOT_ROOT_DIR)/lib/common/ -I $(PREBOOT_ROOT_DIR)/lib/
diff --git a/app/pcie_pif/Makefile b/app/pcie_pif/Makefile
index c365a99..4289855 100644
--- a/app/pcie_pif/Makefile
+++ b/app/pcie_pif/Makefile
@@ -6,7 +6,8 @@
 HL_DEV_ARCH_COMMON := arch/$(BASE_DEV_NAME)/common
 PREBOOT_ROOT_DIR := ../..
 
-CFLAGS += -I$(SPECS_H9_DIR)
+include $(PREBOOT_ROOT_DIR)/$(HL_DEV_ARCH_COMMON)/Makefile
+
 CFLAGS += -I$(SPECS_DIR)
 CFLAGS += -I$(SPECS_EMBEDDED_DIR)
 CFLAGS += -I$(SPECS_EMBEDDED_DIR)/$(BASE_DEV_NAME)
diff --git a/app/pldm_bloader/Makefile b/app/pldm_bloader/Makefile
index 39a4dda..fd2de27 100644
--- a/app/pldm_bloader/Makefile
+++ b/app/pldm_bloader/Makefile
@@ -6,7 +6,8 @@
 HL_DEV_ARCH_COMMON := arch/$(BASE_DEV_NAME)/common
 PREBOOT_ROOT_DIR := ../..
 
-CFLAGS += -I$(SPECS_H9_DIR)
+include $(PREBOOT_ROOT_DIR)/$(HL_DEV_ARCH_COMMON)/Makefile
+
 CFLAGS += -I$(SPECS_DIR)
 CFLAGS += -I$(SPECS_EMBEDDED_DIR)
 CFLAGS += -I$(SPECS_EMBEDDED_DIR)/$(BASE_DEV_NAME)
diff --git a/arch/gaudi3/common/Makefile b/arch/gaudi3/common/Makefile
index 91d2dff..f98ae03 100644
--- a/arch/gaudi3/common/Makefile
+++ b/arch/gaudi3/common/Makefile
@@ -1,3 +1,4 @@
+CFLAGS += -I $(SPECS_H9_DIR)
 
 PPBOOT_SRCS_C += $(shell ls $(HL_DEV_ARCH_COMMON)/common.c)
 PPBOOT_SRCS_C += $(shell ls $(HL_DEV_ARCH_COMMON)/$(HL_TARGET).c)
